### DISCUSSION TITLE
How to close SSE connection from browser

### DISCUSSION THREAD

**Discussion by charliematters** [sergiodxa/remix-utils#328] [Q&A]
I'm trying to use useEventSource, but I notice that when it unmounts it closes the source, but that doesn't trigger the abort controller on the server. Is this by design, or am I doing something wrong?

**Comment by charliematters** [comment]
The context here is that I'm proxying a SSE endpoint on another server, so when the user leaves the page with the server events on, I want to tidy up and close the connection from remix to the other API.

**Comment by sergiodxa** [comment]
The abort controller is not closed by useEventSource, the HTTP server is in charge of that, so it may be a bug in your HTTP server.

**Reply by charliematters** [reply]
You're right - it looks like it's:
* slightly flaky support from the browser - some times it just won't close the connection (as far as I can tell)
* a failing with the vite dev server as seen here: https://github.com/MatthewWid/better-sse/issues/50
